schema/schema.json

```diff
@@ -0,0 +1,355 @@
+                        }
+                    ]
+                },
+                "archive_service": {
+                    "$ref": "#/$definitions/archive_service_schema"
+                },
+                "meta": {
+                    "type": "object"
+                }
+            },
+            "unevaluatedProperties": false,
+            "required": [
+                "client_uuid",
+                "name"
+            ]
+        },
+        "service": {
+            "type": "object",
+            "properties": {
+                "urls": {
+                    "description": "service locator urls",
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "format": "uri"
+                    }
+                },
+                "version": {
+                    "description": "service version",
+                    "type": "string"
+                },
+                "meta": {
+                    "type": "object"
+                }
+            },
+            "required": [
+                "urls"
+            ]
+        },
+        "pscheduler_service_schema": {
+            "type": "object",
+            "properties": {
+                "tools": {
+                    "description": "pscheduler tools installed",
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "tests": {
+                    "description": "pscheduler tests enabled on the host",
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                }
+            },
+            "allOf": [
+                {
+                    "$ref": "#/$definitions/service"
+                }
+            ],
+            "unevaluatedProperties": false,
+            "required": [
+                "tools",
+                "tests"
+            ]
+        },
+        "archive_service_schema": {
+            "type": "object",
+            "properties": {
+                "write_urls": {
+                    "description": "write urls for archive results",
+                    "type": "array",
+                    "items": {
+                        "type": "uri"
+                    }
+                },
+                "type": {
+                    "description": "archiver db type - opensearch, elasticsearch etc.",
+                    "type": "string"
+                },
+                "write_type": {
+                    "description": "write archiver type if different from read archiver",
+                    "type": "string"
+                }
+            },
+            "allOf": [
+                {
+                    "$ref": "#/$definitions/service"
+                }
+            ],
+            "unevaluatedProperties": false
```

Comment:
I think you want to require "type" in addition to "urls" from the base service definition.